BCDA-10190: Update migration guide and _typeFilter guidance with new API functionality (#361)
CMSgov/bcda-static-site: src/_pages/v3/how-to-migrate-v3.md

Reviewer: > This guide helps you migrate from BCDA v1/v2 to version 3 (v3). Key changes include:

Can we just say "migrate from BCDA v1/v2 to v3" ?

## Migration overview

This guide helps you migrate from BCDA v1/v2 to v3. Key changes include:

- New endpoint URL structure
- Unified ExplanationOfBenefit resource for all claims
- Updated extension and code system URLs
- Different resource IDs requiring new matching strategies

> [More about v3]({{ '/v3/introducing-v3' | relative_url }})

Reviewer 2: I like that we cut some of the content with this change, but I think this should be a complete sentence. Maybe something like this sentence which I borrowed from Eric's suggested page metadata:

```suggestion
Learn how BCDA v3 supports better outcomes in our [introduction to v3]({{ '/v3/introducing-v3' | relative_url }}).
```

The link seems out of place on it's own

<img width="1238" height="634" alt="Image" src="https://github.com/user-attachments/assets/05bfafa2-eaca-4cb9-b8ee-5c3cf2c89f6b" />

- Updated extension and code system URLs
- Different resource IDs requiring new matching strategies

> More about v3 [More about v3]({{ '/v3/introducing-v3' | relative_url }})

Reviewer: ```suggestion
[More about v3]({{ '/v3/introducing-v3' | relative_url }})
```

The /auth and /data URLs will remain the same between versions. You won't need to make any changes to how you request a token or download files.

## Changes to claims representation

> In BCDA v2, 3, claims data are still returned in FHIR R4 format, but there are changes to the Patient, Coverage, and ExplanationOfBenefit resources. These changes include:

Reviewer: ```suggestion
In BCDA v3, claims data are still returned in FHIR R4 format, but there are changes to the Patient, Coverage, and ExplanationOfBenefit resources. These changes include:
```

>  - Better conformity with C4BB 2.1.0 profiles

Reviewer 2: ```suggestion
 - Conformity with [CARIN Blue Button Implementation Guide (v2.2.0)](https://hl7.org/fhir/us/carin-bb/STU2.2/en/)
```

Reviewer 2: Per https://jira.cms.gov/browse/BFD-4648

In v2, BCDA differentiates "partially adjudicated" from "fully adjudicated" claims based on the data source rather than the claim status. BCDA's partially adjudicated data (Claim and ClaimResponse resources) are populated with data from the Medicare Shared Systems which often includes data for claims that had been fully processed and paid.

#### How it works in v3

We've extended the API with the [`_typeFilter` parameter]({{ '/v3/filter-claims-data-v3.html#the-typefilter-parameter' | relative_url }}) to filter export data more granularly. Because all claims in v3 are represented by the same resource type (`ExplanationOfBenefit`), use this parameter to specify the System Type _tag, recreating your v2 filtering logic.

> **If you didn’t use the _type parameter in v2 requests,** you received all claims. 

Reviewer 2: This is not 100% accurate because only REACH ACOs have access to partially adjudicated claims data in v2. I think we can update these bold sections just a little bit
```suggestion
**If you received `ExplanationOfBenefit`, `Claim`, and `ClaimResponse` resources in v2 requests,** you received all claims. 
```

Reviewer 2: Omitting the _type param in v2 gave different experiences for REACH ACOs (5 resource types) compared to KCC and MSSP (3 resource types)

Author: ope, yeah good point. updated!

In v3, specify all System Type codes using the _typeFilter parameter.
{% capture sampleRequest %}{% raw %}
GET /api/v3/Patient/$export?_type=ExplanationOfBenefit&_typeFilter=ExplanationOfBenefit?_tag=
  https://bluebutton.cms.gov/fhir/CodeSystem/System-Type|SharedSystem,
  https://bluebutton.cms.gov/fhir/CodeSystem/System-Type|NationalClaimsHistory,
  https://bluebutton.cms.gov/fhir/CodeSystem/System-Type|DDPS
{% endraw %}{% endcapture %}
{% include copy_snippet.html code=sampleRequest %}

> **If you used the _type parameter for `_type=Claim,ClaimResponse` in v2 requests,** you received SharedSystem claims only.

Reviewer 2: This one looks good as-is

In v3, specify the SharedSystem System Type code using the _typeFilter parameter:
{% capture sampleRequest %}{% raw %}
GET /api/v3/Patient/$export?_type=ExplanationOfBenefit&_typeFilter=ExplanationOfBenefit?_tag=
  https://bluebutton.cms.gov/fhir/CodeSystem/System-Type|SharedSystem
{% endraw %}{% endcapture %}
{% include copy_snippet.html code=sampleRequest %}

> **If you used the _type parameter for `_type=ExplanationOfBenefit` for v2 requests,** you received NationalClaimsHistory claims and DDPS claims.

Reviewer 2: This one also looks good as-is

#### How it works in v3

We've extended the API with the [`_typeFilter` parameter]({{ '/v3/filter-claims-data-v3.html#the-typefilter-parameter' | relative_url }}) to filter export data more granularly. Because all claims in v3 are represented by the same resource type (`ExplanationOfBenefit`), use this parameter to specify the System Type _tag, recreating your v2 filtering logic.

**If you didn’t use the _type parameter in v2 requests,** you received all claims. 

In v3, specify all System Type codes using the _typeFilter parameter.
{% capture sampleRequest %}{% raw %}
GET /api/v3/Patient/$export?
  _type=
    ExplanationOfBenefit
  &_typeFilter=
    ExplanationOfBenefit%3F_tag%3Dhttps%3A%2F%2Fbluebutton.cms.gov%2Ffhir%2FCodeSystem%2FSystem-Type%7CSharedSystem%2Chttps%3A%2F%2Fbluebutton.cms.gov%2Ffhir%2FCodeSystem%2FSystem-Type%7CNationalClaimsHistory%2Chttps%3A%2F%2Fbluebutton.cms.gov%2Ffhir%2FCodeSystem%2FSystem-Type%7CDDPS

> {% endraw %}{% endcapture %}

Reviewer: Is this formatting intentional? It breaks how we've been doing it, which might be fine, just want to be sure.

If yes, it might be worth making a future ticket to make it a bit more readable.

<img width="648" height="196" alt="url parameter example where several parameters are broken onto multiple lines and indented" src="https://github.com/user-attachments/assets/73089634-72e0-49d2-9f2c-260325c82439" />

Author: ope, good catch. sorry this was just a test to see how it'd look, compared to the other ones. I pushed this as a result of Greg's comment in a slack thread. Will make sure to make them consistent!

Reviewer 2: I mocked this up quickly following the pattern from the bulk IG and it's even harder to read in my opinion:

<img width="1200" height="626" alt="Image" src="https://github.com/user-attachments/assets/6d335a0b-8b8c-4e12-9d06-c85e4af7f336" />
